client/lib/i18n-utils/switch-locale.js

```diff
@@ -547,15 +547,33 @@ function loadCSS( cssUrl, currentLink ) {
 }
 
 /**
- * Add translations.
+ * Translation data batch strore.
+ *
+ * @type {Array}
+ */
+const _translationsBatch = [];
+
+/**
+ * A throttle wrapper around i18n.addTranslations.
+ *
+ * @param {Object} userTranslations User translations data that will override chunk translations
+ */
+const _addTranslationsBatch = throttle( function ( userTranslations ) {
+	i18n.addTranslations( Object.assign( {}, ..._translationsBatch.splice( 0 ), userTranslations ) );
+}, 100 );
```

Comment:
Does this mean that it is possible for the page to render, and a few ms later, re-render with a different set of translations?

Reply:
Yes, and that's the downside of this approach, but I couldn't think of better workaround for batching the translations data. Maybe the throttle time can be reduced to ~16ms (~1 frame), but that might increase the `addTranslations` calls.